### ISSUE TITLE
test-json fails on ppc64el

### ISSUE THREAD

**Issue by yuwata** [systemd/systemd#21390]
```
/* test_float */
Assertion 'json_variant_is_real(json_variant_by_index(v, 7)) && json_variant_is_integer(json_variant_by_index(v, 7)) && json_variant_integer(json_variant_by_index(v, 7)) == 10' failed at src/test/test-json.c:540, function test_float_match(). Aborting.
```
cc @poettering

**Comment by yuwata** [comment]
Sorry. I merged #21377 too early.

**Comment by poettering** [comment]
i posted PR #21403 that might help tracking down the issues. it mostly outputs some test info.
